src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -45,6 +45,13 @@ internal sealed partial class Http2Connection : HttpConnectionBase, IDisposable
         private long _idleSinceTickCount;
         private int _pendingWriters;
         private bool _lastPendingWriterShouldFlush;
+        private int _freeStreamSlots;
+        private long _freeStreamSlotsExpiration;
+        private const int SlotExpirationPeriod = 1000;
+        // _maxConcurrentStreams == 0 in the beginning because SETTINGS frame is sent asynchronously, so allow requests under the most commont stream limit to proceed.
```

Comment:
I'm confused by this comment, because just below we initialize _maxConcurrentStreams to int.MaxValue.

Reply:
Yeah, it was a mistake. Fixed.